service/discovery/extra/csaf/provider_discover.go

```diff
@@ -62,21 +64,42 @@ func (d *csafDiscovery) handleProvider(lpmd *csaf.LoadedProviderMetadata) (provi
 			SchemaUrl: "https://docs.oasis-open.org/csaf/csaf/v2.0/provider_json_schema.json",
 			Errors:    providerValidationErrors(lpmd.Messages),
 		},
+		Raw: discovery.Raw(pmd),
+	}
+
+	keys := d.discoverKeys(pmd.PGPKeys)
+	providers = append(providers, keys...)
```

Comment:
Maybe append keys in the end of the function where the rest is appended as well. Meanwhile, we could add a comment to `handleProvider` including what ia actually returned (securiyAdvisoryDocuments, keys, ...)